src/main/java/com/ecore/roles/web/RolesApi.java

```diff
@@ -2,18 +2,23 @@
 
 import com.ecore.roles.web.dto.RoleDto;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.util.UriComponentsBuilder;
 
 import java.util.List;
 import java.util.UUID;
 
 public interface RolesApi {
 
     ResponseEntity<RoleDto> createRole(
-            RoleDto role);
+            RoleDto role,
+            UriComponentsBuilder uriBuilder);
```

Comment:
Even though UUID is “URL safe”, but this code can use URI Variables in order to follow a consistency of always use URL template variables.